### PR TITLE
Support more auth types + update HTML report

### PR DESCRIPTION
**Pull request by shelld3v**
Description
---------------

Support multiple auth types

### CODE REVIEW

**Comment by shelld3v** [comment]
@maurosoria

**Comment by shelld3v** [comment]
Don't merge this yet

**Comment by shelld3v** [comment]
@maurosoria Able to merge now!

**Comment by maurosoria** [comment]
This commit breaks the MVC pattern, adding logic model in the argument_parser.py (all the HTTP AUTH classes). That should be encapsulated in a file at '/lib/connection/' and the Requester class should be the responsible of the authentication setup.
https://github.com/maurosoria/dirsearch/pull/819/commits/f41233f345bddb562f4ae25da96bac8b9261b474

* NTLM has some dependencies that should be checked and added to requirements.txt
* NTLM also support Pass The Hash authentication. Could be a good feature for red teamers (Just a suggestion).
* Requests already support BasicAuth (not need for a base64 manual encoding). 
* self.headers shouldn't be touched outside of Requester.py

Sorry for being so demanding ...

Regards,
Mauro

**Comment by shelld3v** [comment]
>NTLM also support Pass The Hash authentication. Could be a good feature for red teamers (Just a suggestion).

It's a vulnerability, not really an authentication method, and I don't see it much in the wild, seem quite useless

File: lib/reports/templates/html_report_template.html [hunk 1]
```diff
@@ -8,11 +8,13 @@
...
 <script>
 function search(app, result) {
-    if (!isNaN(app.searchQuery)) {
-        return result.status == app.searchQuery || result.contentLength == app.searchQuery;
-    }
-    return app.searchQuery.toLowerCase().split(' ').every(v => result.url.toLowerCase().includes(v) ||
-        (result.redirect && result.redirect.toLowerCase().includes(v)));
+    return app.searchQuery.toLowerCase().split(" ").every(
+      (v) => result.url.toLowerCase().includes(v) ||
+      result.contentType.includes(v) ||
```
Comment (wdahlenburg) [inline lib/reports/templates/html_report_template.html:13]: Can you change this to `result.contentType.toLowerCase().includes(v)`?

**Comment by shelld3v** [comment]
@maurosoria

**Pull request merged**